src/Microsoft.ML.OnnxTransform/OnnxCatalog.cs

```diff
@@ -10,24 +10,54 @@ namespace Microsoft.ML
         /// <param name="catalog">The transform's catalog.</param>
         /// <param name="modelFile">The path of the file containing the ONNX model.</param>
-        /// <param name="outputColumnNames">The input columns.</param>
-        /// <param name="inputColumnNames">The output columns resulting from the transformation.</param>
+        /// <param name="gpuDeviceId">Optional GPU device ID to run execution on, <see langword="null" /> to run on CPU.</param>
+        /// <param name="fallbackToCpu">If GPU error, raise exception or fallback to CPU.</param>
         public static OnnxScoringEstimator ApplyOnnxModel(this TransformsCatalog catalog,
             string modelFile,
-            string[] outputColumnNames,
-            string[] inputColumnNames)
-        => new OnnxScoringEstimator(CatalogUtils.GetEnvironment(catalog), outputColumnNames, inputColumnNames, modelFile);
+            int? gpuDeviceId = null,
+            bool fallbackToCpu = false)
+        => new OnnxScoringEstimator(CatalogUtils.GetEnvironment(catalog), modelFile, gpuDeviceId, fallbackToCpu);
 
         /// <summary>
-        /// Initializes a new instance of <see cref="OnnxScoringEstimator"/>.
+        /// Applies a pre-trained Onnx model.
         /// </summary>
         /// <param name="catalog">The transform's catalog.</param>
-        /// <param name="transformer">The ONNX transformer.</param>
-        public static OnnxScoringEstimator ApplyOnnxModel(this TransformsCatalog catalog, OnnxTransformer transformer)
-            => new OnnxScoringEstimator(CatalogUtils.GetEnvironment(catalog), transformer);
+        /// <param name="modelFile">The path of the file containing the ONNX model.</param>
+        /// <param name="outputColumnName">The input column.</param>
+        /// <param name="inputColumnName">The output column resulting from the transformation.</param>
+        /// <param name="gpuDeviceId">Optional GPU device ID to run execution on, <see langword="null" /> to run on CPU.</param>
+        /// <param name="fallbackToCpu">If GPU error, raise exception or fallback to CPU.</param>
+        public static OnnxScoringEstimator ApplyOnnxModel(this TransformsCatalog catalog,
+            string modelFile,
+            string outputColumnName,
+            string inputColumnName,
+            int? gpuDeviceId = null,
+            bool fallbackToCpu = false)
+        => new OnnxScoringEstimator(CatalogUtils.GetEnvironment(catalog), new[] { outputColumnName }, new[] { inputColumnName }, modelFile, gpuDeviceId, fallbackToCpu);
+
+        /// <summary>
+        /// Applies a pre-trained Onnx model.
+        /// </summary>
+        /// <param name="catalog">The transform's catalog.</param>
+        /// <param name="modelFile">The path of the file containing the ONNX model.</param>
+        /// <param name="outputColumnNames">The input columns.</param>
+        /// <param name="inputColumnNames">The output columns resulting from the transformation.</param>
```

Comment:
can you change their description? you put input column for outputColumnName and vice versa.